torch/csrc/autograd/input_buffer.cpp

```diff
@@ -86,13 +86,12 @@ static void accumulate(
     } else {
       buffer[pos] = var + old_var;
     }
+  } else if (
+      old_var.is_contiguous() && old_var.use_count() == 1 &&
```

Comment:
I don't think we want to take this path when both `var` and `old_var` are sparse, who even knows if we have working inplace sparse-sparse addition, and in any case it's a conceptually weird operation, so you might want to push this case to the `else` branch.

Reply:
If `old_var` is sparse it will get hooked by the first part of the conditional. (Which as a `!var.is_sparse()` check on its steal check.) Maybe I should add a cautionary comment?

Reply:
Ah yeah that would be good, thanks for checking!